#!/usr/bin/env python

PACKAGE = "aqc_input_dynamic_reconfigure_vel"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add( "east_rate_setpoint" , double_t , 1 , "east (+x) linear velocity setpoint (m/s)" , 0.0 , -3.0 , 3.0 )
gen.add( "north_rate_setpoint" , double_t , 2 , "north (+y) linear velocity setpoint (m/s)" , 0.0 , -3.0 , 3.0 )
gen.add( "up_rate_setpoint" , double_t , 4 , "up (+z) linear velocity setpoint (m/s)" , 0.0 , -2.0 , 2.0 )
gen.add( "yaw_rate_setpoint" , double_t , 8 , "yaw angular velocity setpoint (rad/s)" , 0.0 , -1.5 , 1.5 )

arm_cli_enum = gen.enum( [gen.const("disarm", int_t, 0, "disarm vehicle"),
		    	          gen.const("arm", int_t, 1, "arm vehicle")], "enum to act as arm_client" )

cfm_cli_enum = gen.enum( [gen.const("takeoff", int_t, 0, "takeoff flight mode"),
                          gen.const("land", int_t, 1, "land flight mode"),
                          gen.const("return_to_land", int_t, 2, "return to land flight mode"),
		    	          gen.const("hold", int_t, 3, "hold (loiter) flight mode"),
                          gen.const("offboard", int_t, 4, "offboard flight mode"),
                          gen.const("manual", int_t, 5, "manual flight mode"),
                          gen.const("position", int_t, 6, "position flight mode"),
                          gen.const("stabilized", int_t, 7, "stabilized flight mode"),
                          gen.const("acro", int_t, 8, "acro flight mode"),
                          gen.const("altitude", int_t, 9, "altitude flight mode")], "enum to act as cfm_client" )

gen.add( "arm_request" , int_t , 16 , "client to arm or disarm vehicle" , 0 , 0 , 1 , edit_method=arm_cli_enum )
gen.add( "cfm_request" , int_t , 32 , "client to change vehicle flight mode" , 3 , 0 , 9 , edit_method=cfm_cli_enum )

exit( gen.generate(PACKAGE, "aqc_input_dynamic_reconfigure_vel", "AqcInputClientVel"))
